pkg/resource/resource.go

```diff
@@ -602,3 +605,33 @@ func checkForImagePullBackOff(obj *unstructured.Unstructured, namespace string)
 
 	return fmt.Errorf("unexpected Phase of Pods in DameonSet: %s", obj.GetName())
 }
+
+func sendNodesMetrics(obj *unstructured.Unstructured, crName string) {
+	kind := obj.GetKind()
+	if kind != "DaemonSet" && kind != "Deployment" {
+		return
+	}
+
+	objKey := types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+	getPodsFunc := lifecycle.GetPodFromDaemonSet
+	if kind == "Deployment" {
+		getPodsFunc = lifecycle.GetPodFromDeployment
+	}
+
+	pl := getPodsFunc(objKey)
+	nodesNames := []string{}
```

Comment:
```suggestion
	nodesNames := make([]string, 0, len(pl.Items))
```

Reply:
don't see much difference, mainly when the underlying array allocation is performed and at what size, which is, in our case, insignificant